util/file_reader_writer.cc

```diff
@@ -124,6 +124,17 @@ Status WritableFileWriter::Append(const Slice& data) {
     writable_file_->PrepareWrite(static_cast<size_t>(GetFileSize()), left);
   }
 
+  // See whether we need to enlarge the buffer to avoid the flush
+  if (buf_.Capacity() - buf_.CurrentSize() < left) {
+    for (size_t desired_capacity = buf_.Capacity();
+         desired_capacity < max_buffer_size_; desired_capacity *= 2) {
+      if (desired_capacity - buf_.CurrentSize() >= left) {
```

Comment:
I think we should cut desired_capacity to max_buffer_size_ if desired_capacity > max_buffer_size_ before this line.